### ISSUE TITLE
Qubes Update widget reports updates available when none are available

### ISSUE THREAD

**Issue by n0sign4l** [QubesOS/qubes-issues#4667]
### Qubes OS version:

Qubes release 4.0 (R4.0)

### Affected component(s): 

Qubes update GUI.
I open the qubes-update-gui from the icon than i see that i have to update some of the things listed, I select them than click on next, but all of them get SKIP nothing to do

---

### Steps to reproduce the behavior:

1. open `qubes-update-gui`
2. select all the things you have to update
3. click `next`
4. click `details`
5. see that everything that got selected say: 'SKIP (nothing to do)'
6. on the top-right corner I still have the orange icon that says me that I have to update something.

### Expected behavior:

Everything should update correctly

### Actual behavior:

Things do not get updated

**Comment by marmarek** [comment]
This is already fixed by https://github.com/QubesOS/qubes-issues/issues/2451 (update in current-testing repository)

**Comment by n0sign4l** [comment]
I still have issue with this, I have updated right now to the latest version of Qubes, but for example fedora-26 and whonix-gw do NOT get updated. I don't get anymore SKIP (nothing to do), now disp machines start and a lot of things are done but the result is the same, once everything is done I still have the yellow icon on the top-right corner and it says that I have to update those VMs.

**Comment by marmarek** [comment]
If you expand "details" in the updater, do you see some updates installed, or some error, or else?

**Comment by andrewdavidwong** [comment]
> I still have issue with this, I have updated right now to the latest version of Qubes

"The latest version of Qubes" is often used to mean the latest _stable_ updates, but not necessarily all the latest _testing_ updates (and not necessarily the fix for #2451 in particular). Can you confirm that you have the actual fix for #2451? (Instructions to get it are available from the @qubesos-bot comments on #2451.)

**Comment by n0sign4l** [comment]
This is the output
`Updating fedora-26

fedora-26:
      ----------

Updating whonix-gw

whonix-gw:
      ----------
      retcode:
          0
      stderr:
          Error running 'pkg.upgrade': E: Release file for http://deb.whonix.org/dists/jessie/InRelease is expired (invalid since 9d 13h 5min 31s). Updates for this repository will not be applied.
          E: dpkg was interrupted, you must manually run 'dpkg --configure -a' to correct the problem.
      stdout:
`

Leonardo 'n0sign4l' Porpora

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐
Il domenica, gennaio 6, 2019 11:18 PM, Marek Marczykowski-Górecki <notifications@github.com> ha scritto:

**Comment by n0sign4l** [comment]
Any update??

**Comment by marmarek** [comment]
>  Error running 'pkg.upgrade': E: Release file for http://deb.whonix.org/dists/jessie/InRelease is expired (invalid since 9d 13h 5min 31s). Updates for this repository will not be applied.

This looks to be Whonix 13, which is not supported anymore. See https://www.qubes-os.org/news/2018/08/24/whonix-13-approaching-eol/

**Comment by n0sign4l** [comment]
I'm not sure that this is the only one problem cause also fedora-26 does not get updated...
And I'm not able also to update Whonix from 13 to 14 as if I click to open the terminal of Whonix VM it does NOT come up... Please help

Leonardo 'n0sign4l' Porpora

**Comment by marmarek** [comment]
> I'm not sure that this is the only one problem cause also fedora-26 does not get updated...

Same here: https://www.qubes-os.org/news/2018/05/23/fedora-26-and-debian-8-approaching-eol/

> And I'm not able also to update Whonix from 13 to 14 as if I click to open the terminal of Whonix VM it does NOT come up... Please help

See https://www.whonix.org/wiki/Upgrading_Whonix_13_to_Whonix_14#Download_New_Templates

**Comment by andrewdavidwong** [comment]
Reopening and increasing priority because multiple users (including me) have also been experiencing this, even after updates in #2451 hit stable.

**Comment by andrewdavidwong** [comment]
I wonder if this is actually a duplicate of #4593 (if #4593 is not actually limited to the `fedora-29` template).

**Comment by n0sign4l** [comment]
I still have this issue too.... I can confirm that the problem seems limited with fedora @marmarek @andrewdavidwong

**Comment by nimrodl** [comment]
I have a new qubes install and I'm seeing this too. The tray icon shows up, but when the updater gui is launched, none of the domains are checked as requiring updates. When checking each one manually and running the updates it reports back that there is nothing to do. But the tray icon still reports that one qube has pending updates.

**Comment by marmarek** [comment]
Does it still happen? The above linked update contains related fix.

**Comment by n0sign4l** [comment]
No it didn't happened anymore for me

**Comment by andrewdavidwong** [comment]
Closing this as resolved. If anyone is still affected by this issue after applying the above fix, please leave a comment, and we'll be happy to reopen this. Thank you.

**Comment by andrewdavidwong** [comment]
Reopening due to apparent regression reported in #5683.

**Comment by de-souza** [comment]
This happens to me specifically with the packages ffmpeg-libs and/or vlc from the RPM Fusion free repository, in the Fedora 31 TemplateVM. The issue disappears when I remove these packages.

**Comment by pabpas** [comment]
This still happens with the Fedora 32 TemplateVM. As workaround I have manually removed libdav1d.x86_64 (0.5.1?) and seems to be solved.

`rpm -ev --nodeps libdav1d.x86_64`

**Comment by andrewdavidwong** [comment]
QubesOS/qubes-doc@f539ae1 doesn't appear to be related to this issue. Must have been a typo.

**Comment by andrewdavidwong** [comment]
Is anyone still affected by this issue?

**Comment by de-souza** [comment]
> [QubesOS/qubes-doc@f539ae1](https://github.com/QubesOS/qubes-doc/commit/f539ae1) doesn't appear to be related to this issue. Must have been a typo.

Sorry, this would have deserved an explanation. This issue appeared for me until I added the `rpmfusion-free-updates` and `rpmfusion-nonfree-updates` repositories. After I did, the update widget started to behave normally.

**Comment by andrewdavidwong** [comment]
> > [QubesOS/qubes-doc@f539ae1](https://github.com/QubesOS/qubes-doc/commit/f539ae1) doesn't appear to be related to this issue. Must have been a typo.
> 
> Sorry, this would have deserved an explanation. This issue appeared for me until I added the `rpmfusion-free-updates` and `rpmfusion-nonfree-updates` repositories. After I did, the update widget started to behave normally.

I wonder why, though. For some reason, the VM though there were updates available, and only once it was able to check `rpmfusion-free-updates` and `rpmfusion-nonfree-updates` was it able to see that none were available? Why would it think that updates were available in the first place, though? Seems like there must be a bug somewhere.

**Comment by marmarek** [comment]
That indeed may be an explanation. If you have some package installed that depends on something from rpmfusion repo (for example you installed it in the past with rpmfusion enabled temporarily), then the normal update process won't be able to update said package with rpmfusion disabled (because of missing dependency). In that case, there it will appear as some updates are always available (because they are), but won't be installed.
I think I've seen similar issue with some chromium-related package.

**Comment by andrewdavidwong** [comment]
I bet that was exactly it.
